File: Quick Find/QuickFind.py
class QuickFind:
    
    def __init__(self, N):
        self.ids = [k for k in range(N)]
        self.size = N
    
    def find(self, i):
        return self.ids[i]

    def union(self, i, j):
        iID = self.find(i)
        jID = self.find(j)
        if (iID != jID):
            for k in range(self.size):
                if (self.ids[k] == iID):
                    self.ids[k] = jID

N = int(input())
M = int(input())
lines = [input().split() for _ in range(M)]
QF = QuickFind(N)
for line in lines:
    if line[0]=='F':
        print(QF.find(int(line[1])))
    elif line[0]=='U':
        QF.union(int(line[1]),int(line[2]))

File: Union Find/Quick Find/QuickFind.cpp
#include <bits/stdc++.h>
using namespace std;

class QuickFind{
    vector<int> ids;
    int size;

    public:
        QuickFind(int N){
            ids = vector<int>(N);
            for (int i=0;i<N;i++)
                ids[i] = i;
            size = N;
        }

        int find(int i) {
            return ids[i];
        }

        void qunion(int i, int j){
            int iID = find(i);
            int jID = find(j);
            if (iID != jID)
                for (int k=0;k<size;k++)
                    if (ids[k] == iID)
                        ids[k] = jID;
            
        }
};

int main() {
    int N,M;
    cin>>N;
    cin>>M;
    QuickFind qf = QuickFind(N);
    for (int i=0;i<M;i++){
        char c;
        cin>>c;
        if (c=='F'){
            int x;
            cin>>x;
            cout<<qf.find(x)<<endl;
        } else if (c=='U') {
            int x,y;
            cin>>x>>y;
            qf.qunion(x,y);
        }
    }
}